Modules/selectmodule.c

```diff
@@ -554,9 +555,17 @@ poll_poll(pollObject *self, PyObject *args)
             return NULL;
         }
 
-        deadline = _PyTime_GetMonotonicClock() + timeout;
+        if (timeout >= 0)
```

Comment:
Style nit: Per PEP 7, we now use the following style in new C code:

```c
if (spam) {
    return 42;
}
````

instead of

```c
if (spam)
    return 42;
````